internal/app/app.go

```diff
@@ -53,32 +49,34 @@ type options struct {
 type option func(*options)
 
 type authorizerer interface {
-	IsSenderAllowed(context.Context, string, dbus.Sender) error
+	IsSenderAllowed(string, dbus.Sender) error
 }
 type proxyApplier interface {
-	Apply(context.Context, string, string, string, string, string, string) error
+	Apply(string, string, string, string, string, string) error
 }
 
 // Apply is a function called via D-Bus to apply the system proxy settings.
 func (b *proxyManagerBus) Apply(sender dbus.Sender, http, https, ftp, socks, no, mode string) *dbus.Error {
-	// Methods calls spin up separate goroutines, so ensure we don't run them in parallel
-	b.mu.Lock()
-	defer b.mu.Unlock()
-
 	log.Debugf("Sender %s called Apply(%q, %q, %q, %q, %q, %q)", sender, http, https, ftp, socks, no, mode)
 
-	// We need to cancel the context in a deferred function to get the final
-	// state of the error variable, and to let the main thread know it's safe to quit.
+	// Application was already asked to quit, so return an error without applying anything
+	if b.l.QuitRequested() {
+		return dbus.MakeFailedError(errors.New("application exit requested, cannot apply proxy settings"))
```

Comment:
I’m a little bit not at ease with this: it means that we could get there, BUT, a new request has just come. So, we can proceed, no?